### PR TITLE
Backport ansible-2.2 fixes

### LINKED ISSUE #2479: TASK openshift_node : Configure Proxy Settings - Fails
We got this when deploying 2 Master nodes cluster from Ansible into Atomic Hosts.
Even if we define or not proxy cluster vars, it fails. We don't use proxy in our env.

OUTPUT:

```
TASK [openshift_node : Configure Proxy Settings] *******************************
task path: /drake/openshift-ansible/roles/openshift_node/tasks/systemd_units.yml:51
fatal: [master2.demo.<mydomain>.com]: FAILED! => {
    "failed": true, 
    "msg": "'dict object' has no attribute 'http_proxy'"
}

RUNNING HANDLER [openshift_node : restart openvswitch] *************************

RUNNING HANDLER [openshift_node : restart node] ********************************
    to retry, use: --limit @/drake/openshift-ansible/playbooks/byo/config.retry

PLAY RECAP *********************************************************************
localhost                  : ok=12   changed=7    unreachable=0    failed=0   
master1.demo.<mydomain>.com : ok=169  changed=20   unreachable=0    failed=1   
master2.demo.<mydomain>.com : ok=220  changed=41   unreachable=0    failed=1
```

### PR DESCRIPTION
**Pull request by sdodson**
Fixes Bug 1386654

Fixes #2479 
Backports #2518 
Backports #2630

Per https://bugzilla.redhat.com/show_bug.cgi?id=1386654#c5 "with_ clauses ALWAYS
evaluate before the when clause, otherwise we cannot use item and condition loop
iterations. So this is not a bug, just how ansible works. The default and
ternary filters can be used to avoid the undefined error."

### CODE REVIEW

**Comment by sdodson** [comment]
Probably need to rebase this onto master then backport to 3.3 and 3.2, but running CI jobs real quick.

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
@abutcher PTAL, also any other Ansible 2.2 compat items we may have missed?

**Comment by abutcher** [comment]
@sdodson Templated with_items already backported so I think the only thing left would be #2630.

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
`make ci` doesn't exist on this branch, that's why this one is failing.  @abutcher or @detiber PTAL need to get this in today if possible.

File: roles/openshift_node/tasks/systemd_units.yml [hunk 1]
```diff
@@ -56,12 +56,12 @@
     create: true
   with_items:
     - regex: '^HTTP_PROXY='
-      line: "HTTP_PROXY={{ openshift.common.http_proxy }}"
+      line: "HTTP_PROXY={{ openshift.common.http_proxy | default('') }}"
```
Comment (gundalow) [inline roles/openshift_node/tasks/systemd_units.yml]: Does this fix the issues reported in https://bugzilla.redhat.com/show_bug.cgi?id=1386654

**Pull request merged**